src/MagnumPlugins/UfbxImporter/UfbxImporter.cpp

```diff
@@ -905,23 +916,23 @@ Containers::Optional<MaterialData> UfbxImporter::doMaterial(UnsignedInt id) {
                 }
             }
 
-            if (mapping.texture != (MaterialAttribute)0) {
+            if (mapping.texture != MaterialAttribute(0)) {
                 ufbx_texture *texture = colorMap.texture;
-                if (texture && colorMap.texture_enabled && !addedAttributes[(UnsignedInt)mapping.texture]) {
-                    addedAttributes.set((UnsignedInt)mapping.texture, true);
-                    arrayAppend(attributes, {mapping.texture, (UnsignedInt)texture->typed_id});
+                if (texture && colorMap.texture_enabled && !addedAttributes[UnsignedInt(mapping.texture)]) {
+                    addedAttributes.set(UnsignedInt(mapping.texture), true);
+                    arrayAppend(attributes, {mapping.texture, UnsignedInt(texture->typed_id)});
 
-                    if (mapping.textureMatrix != (MaterialAttribute)0 && texture->has_uv_transform) {
+                    if (mapping.textureMatrix != MaterialAttribute(0) && texture->has_uv_transform) {
 
                         /* Texture matrix should be connected to texture */
-                        CORRADE_INTERNAL_ASSERT(!addedAttributes[(UnsignedInt)mapping.textureMatrix]);
-                        addedAttributes.set((UnsignedInt)mapping.textureMatrix, true);
+                        CORRADE_INTERNAL_ASSERT(!addedAttributes[UnsignedInt(mapping.textureMatrix)]);
+                        addedAttributes.set(UnsignedInt(mapping.textureMatrix), true);
 
                         const ufbx_matrix &mat = texture->uv_to_texture;
                         Matrix3 value = {
-                            { (Float)mat.m00, (Float)mat.m10, 0.0f },
-                            { (Float)mat.m01, (Float)mat.m11, 0.0f },
-                            { (Float)mat.m03, (Float)mat.m13, 0.0f },
+                            { Float(mat.m00), Float(mat.m10), 0.0f },
+                            { Float(mat.m01), Float(mat.m11), 0.0f },
+                            { Float(mat.m03), Float(mat.m13), 0.0f },
```

Comment:
Oops, I think the slicing would be wrong for the exact same reason as it would drop the `1` in the bottom right corner (which doesn't even exist in a 4x3 matrix with implicit `0 0 0 1` row).

Reply:
If the original `ufbx_matrix` in `uv_to_texture`  has `1.0` at `[2][2]` (which I suppose it has?), the slicing would preserve it.

Reply:
I'm treating it as a 2D homogeneous matrix, let's say (perspective on the Y axis, imagine that!), it's just that the way it's stored as the first 3 columns/rows of a 3D homogeneous matrix makes it look a bit suspicious.